src/main/java/edu/utdallas/davisbase/compiler/Compiler.java

```diff
@@ -689,4 +728,87 @@ private void validateNotCatalogTable(String tableName) throws CompileException {
     }
   }
 
+  /**
+   * Throws CompileException if column constraint is unique but trying to insert non unique value
+   * @param tableName name of table
+   * @param columnName name of column with constraint
+   * @param value value trying to insert
+   * @throws StorageException
+   * @throws IOException
+   * @throws CompileException
+   */
+  public void validateUniqueness(String tableName, String columnName, Object value)throws StorageException, IOException, CompileException{
+    //TODO: Add index logic
+    if(isUnique(tableName, columnName)){
+      byte colIndex = validateIsDavisBaseColumnWithinTable(tableName, columnName);
+      DataType colType = getColumnType(tableName, columnName);
+      TableFile table = context.openTableFile(tableName);
+      final String UNIQUENESS_EXCEPTION = "Invalid insert. Column " + columnName + " has uniqueness constraint";
+      while(table.goToNextRow()){
+        if(colType==DataType.TINYINT && castNonNull(table.readTinyInt(colIndex)).equals(value)){
+          throw new CompileException(UNIQUENESS_EXCEPTION);
+        }
+        else if(colType==DataType.SMALLINT && castNonNull(table.readSmallInt(colIndex)).equals(value)){
+          throw new CompileException(UNIQUENESS_EXCEPTION);
+        }
+        else if(colType==DataType.INT && castNonNull(table.readInt(colIndex)).equals(value)){
+          throw new CompileException(UNIQUENESS_EXCEPTION);
+        }
+        else if(colType==DataType.BIGINT && castNonNull(table.readBigInt(colIndex)).equals(value)){
+          throw new CompileException(UNIQUENESS_EXCEPTION);
+        }
+        else if(colType==DataType.FLOAT && castNonNull(table.readFloat(colIndex)).equals(value)){
```

Comment:
```suggestion
        else if(colType==DataType.FLOAT && Objects.equals(table.readFloat(colIndex), value)){
```

The value being read could legitimately be `null`.